### PR TITLE
[SE-0206] Update CHANGELOG.md

### PR DESCRIPTION
**Pull request by lorentey**
Describe hashing improvements scheduled for release in 4.2.

### CODE REVIEW

**Comment by lorentey** [comment]
@swift-ci smoke test

**Comment by lorentey** [comment]
@swift-ci smoke test

File: CHANGELOG.md [hunk 1]
```diff
@@ -61,6 +61,73 @@ Swift 5.0
...
+* [SE-0206][]
+
+    The standard library now uses a high-quality, randomly seeded, universal
+    hash function, represented by the new public `Hasher` struct.
+  
+    “Random seeding” means that `hashValue` properties will return different
+    values on each execution of a Swift program. This is an important tool for
+    improving the reliability of the standard library’s hashing collections,
+    `Set` and `Dictionary`. In particular, random seeding enables better
+    protection against (accidental or deliberate) hash-flooding attacks.
```
Comment (milseman) [inline CHANGELOG.md]: Word smithing possibility:

“Random seeding” varies the result of `hashValue` on each execution of a Swift program, improving the reliability of the standard library's hashed collections such as `Set` and `Dictionary`. In particular, random seeding enables better protection against (accidental or deliberate) hash-flooding attacks.

**Comment by lorentey** [comment]
@swift-ci smoke test and merge

**Comment by lorentey** [comment]
@swift-ci smoke test and merge

**Pull request merged**